packages/js-drive/lib/identity/masternode/synchronizeMasternodeIdentitiesFactory.js

```diff
@@ -0,0 +1,141 @@
+const Identifier = require('@dashevo/dpp/lib/identifier/Identifier');
+const { hash } = require('@dashevo/dpp/lib/util/hash');
+
+/**
+ *
+ * @param {DashPlatformProtocol} transactionalDpp
+ * @param {DriveStateRepository|CachedStateRepositoryDecorator} stateRepository
+ * @param {SimplifiedMasternodeList} simplifiedMasternodeList
+ * @param {Identifier} masternodeRewardSharesContractId
+ * @param {handleNewMasternode} handleNewMasternode
+ * @param {handleUpdatedPubKeyOperator} handleUpdatedPubKeyOperator
+ * @param {splitDocumentsIntoChunks} splitDocumentsIntoChunks
+ * @return {synchronizeMasternodeIdentities}
+ */
+function synchronizeMasternodeIdentitiesFactory(
+  transactionalDpp,
+  stateRepository,
+  simplifiedMasternodeList,
+  masternodeRewardSharesContractId,
+  handleNewMasternode,
+  handleUpdatedPubKeyOperator,
+  splitDocumentsIntoChunks, // TODO - DI!!!!
+) {
+  let lastSyncedCoreHeight = 0;
+
+  /**
+   * @typedef synchronizeMasternodeIdentities
+   * @param {number} coreHeight
+   * @return Promise<void>
+   */
+  async function synchronizeMasternodeIdentities(coreHeight) {
+    let documentsToCreate = [];
+    let documentsToDelete = [];
+
+    let newMasternodes = [];
+    let masternodesWithNewPubKeyOperator = [];
+
+    let previousMNList = [];
+
+    if (lastSyncedCoreHeight > 0) {
+      previousMNList = simplifiedMasternodeList.getStore()
+        .getSMLbyHeight(lastSyncedCoreHeight)
+        .mnList;
+    }
+
+    const currentMNList = simplifiedMasternodeList.getStore()
+      .getSMLbyHeight(coreHeight)
+      .mnList;
+
+    if (lastSyncedCoreHeight === 0) {
+      // Create identities for all masternodes on the first sync
+      newMasternodes = simplifiedMasternodeList.getStore().getCurrentSML().mnList;
+    } else {
```

Comment:
You can simplify conditions here:

```suggestion
    if (lastSyncedCoreHeight > 0) {
      previousMNList = simplifiedMasternodeList.getStore()
        .getSMLbyHeight(lastSyncedCoreHeight)
        .mnList;
        
      currentMNList = simplifiedMasternodeList.getStore()
      .getSMLbyHeight(coreHeight)
      .mnList;
    } else {
      previousMNList = []
      currentMNList  = simplifiedMasternodeList.getStore().getCurrentSML().mnList
    }
```